http4s/server/src/main/scala/endpoints4s/http4s/server/Assets.scala

```diff
@@ -0,0 +1,167 @@
+package endpoints4s.http4s.server
+
+import java.net.URL
+
+import cats.effect.{Blocker, ContextShift}
+import cats.implicits._
+import endpoints4s.algebra.Documentation
+import endpoints4s.{Valid, algebra}
+import fs2.io._
+import org.http4s.headers._
+import org.http4s._
+
+trait Assets extends algebra.Assets with EndpointsWithCustomErrors {
+  val DefaultBufferSize = 10240
+
+  case class AssetRequest(
+      assetPath: AssetPath,
+      isGzipSupported: Boolean,
+      ifModifiedSince: Option[HttpDate]
+  )
+
+  case class AssetPath(path: Seq[String], digest: Option[String], name: String)
+
+  sealed trait AssetResponse
+  object AssetResponse {
+    case object NotFound extends AssetResponse
+    case class Found(
+        data: fs2.Stream[Effect, Byte],
+        contentLength: Long,
+        lastModified: Option[HttpDate],
+        mediaType: Option[MediaType],
+        isGzipped: Boolean,
+        expired: Boolean
+    ) extends AssetResponse
+  }
+
+  override def assetSegments(
+      name: String,
+      docs: Documentation
+  ): Path[AssetPath] = {
+    case p :+ s =>
+      val i = s.lastIndexOf('-')
+      val assetPath =
+        if (i > 0) {
+          val (resourceName, digest) = s.splitAt(i)
+          AssetPath(p, Some(digest.drop(1)), resourceName)
```

Comment:
That would indeed work to solve the most common issue. Another solution would be to delegate the interpretation of the digest in the asset path to the `toResourceUrl` function? There we could try to extract the digest and file name. If the filename does not exist, it means that the digest was part of the filename, and this resource could be served instead.

Reply:
You have a good point. Another possibility would be to drop the digest functionality altogether from the http4s interpreter (it is not required by the algebra), and postpone solving this problem (so users won't face unexpected behaviour)? The http4s has caching functionality through If-Modified-Since header implemented already anyway

Reply:
Sure, I’m fine with the solution you propose :)

Reply:
Removed in https://github.com/endpoints4s/endpoints4s/pull/607/commits/98094555d8de11a6a57af107945842509d6296cd

Reply:
Removed https://github.com/endpoints4s/endpoints4s/pull/607/commits/884b8ae642daad920ca649377a4a96ee6a34095e